### DISCUSSION TITLE
GraphQLTest: No Root resolvers for mutation/query type when running unit test

### DISCUSSION THREAD

**Discussion by AndreKunitz** [graphql-java-kickstart/graphql-spring-boot#416] [Q&A] [answered]
**Describe the bug**

I've createted two GrahpQL Schemas separately for two diferent types.
One type is only used in Queries, and the other one only used in Mutations.
Using a package by feature/domain approach, instead of a package by layer.

Then, I've implemented `GrahpQLQueryResolver` and `GrahpQLMutationResolver` for the respective methods.
Everything works normal. I can run queries and mutations in `GraphiQL`.

The problem occurs with unit testing using `GraphQLTest`.

Whenever I run unit test for QueryResolver, I got and error: 

`graphql.kickstart.tools.SchemaClassScannerError: No Root resolvers for mutation type 'Mutation' found! `

The same occurs when unit testing MutationResolver:

`graphql.kickstart.tools.SchemaClassScannerError: No Root resolvers for query type 'Query' found!  Provide one or more graphql.kickstart.tools.GraphQLQueryResolver to the builder.`

Query test runs normally if I remove type Mutation from the schema.
Mutation test runs normally if I remove type Query from the schema.

What I'm missing here? 

**To Reproduce**

[Demo](https://github.com/AndreKunitz/issue-graphql) project that reproduces the behavior.

**Comment by AndreKunitz** [comment]
Packaging by layer solves the problem. But I'd like continue using the first approach. 

Package by feature
![screenshot_19](https://user-images.githubusercontent.com/28653182/83407480-2ca4eb80-a3e7-11ea-86ba-83315fddbc27.png)
Package by layer
![screenshot_20](https://user-images.githubusercontent.com/28653182/83407484-329acc80-a3e7-11ea-9f65-4145c514fba2.png)

**Comment by oliemansm** [comment, marked as answer]
Have you tried adding `@ComponentScan("me.github.andrekunitz.issueexemple")` to your test classes when packaged by feature? There's a `@ComponentScan` without specifying any packages on the `@GraphQLTest` annotation, which might result in it scanning only the package that test class is defined in, instead of the root package.

**Comment by oliemansm** [comment]
@AndreKunitz Did my previous suggestion fix the issue for you?

**Comment by AndreKunitz** [comment]
Sorry for the late response.

Indeed, your suggestion fixed the issue. 
I'll be updating the demo project with the solution. 

Thanks a lot, @oliemansm.
